packages/core/src/libraries/domain.ts

```diff
@@ -71,6 +71,12 @@ export const createDomainLibrary = (queries: Queries) => {
     const { hostnameProviderConfig } = SystemContext.shared;
     assertThat(hostnameProviderConfig, 'domain.not_configured');
 
+    const { blockedDomains } = hostnameProviderConfig;
+    assertThat(
+      !(blockedDomains ?? []).some((domain) => hostname.endsWith(domain)),
```

Comment:
`endsWith` may produce false positive. say we have `foo.com` configured and the domain is `barfoo.com`. this should also be covered in unit tests